src/web/app/user-join-page.component.ts

```diff
@@ -81,9 +91,17 @@ export class UserJoinPageComponent implements OnInit {
    * Joins the course.
    */
   joinCourse(): void {
-
     this.courseService.joinCourse(this.key, this.entityType, this.institute, this.mac).subscribe(() => {
-      this.navigationService.navigateByURL(this.router, `/web/${this.entityType}`);
+      if (this.sampleCourseId && this.timezone) {
+        this.courseService.updateCourse(this.sampleCourseId, {
+          courseName: 'Sample Course 101',
```

Comment:
Be careful. You're renaming the course here, regardless of what the input course ID is (who's to guarantee that it corresponds to sample course?).